src/Datadog.Trace/Agent/Transports/HttpClientRequest.cs

```diff
@@ -22,10 +22,10 @@ public void AddHeader(string name, string value)
             _request.Headers.Add(name, value);
         }
 
-        public async Task<IApiResponse> PostAsync(Span[][] traces, FormatterResolverWrapper formatterResolver)
+        public async Task<IApiResponse> PostAsync(ArraySegment<byte> traces)
         {
             // re-create HttpContent on every retry because some versions of HttpClient always dispose of it, so we can't reuse.
-            using (var content = new TracesMessagePackContent(traces, formatterResolver))
+            using (var content = new BinaryTracesMessagePackContent(traces))
```

Comment:
Can we use the built-in `ByteArrayContent` now instead of a custom one? If we do this, don't forget to add the content type (which I think is all we do now in `BinaryTracesMessagePackContent)`.
```suggestion
            using (var content = new ByteArrayContent(traces.Array, traces.Offset, traces.Count))
            {
                Headers.ContentType = new MediaTypeHeaderValue("application/msgpack");
```

Reply:
Oh, you're right 👍

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/1151/commits/036ae23d5ace2c8e19264ac624a2c0ec63de27e6

Reply:
Are we still adding the content type header somewhere?

Reply:
If not, we can probably use `HttpClientRequestFactory` to add it to `HttpClient.DefaultRequestHeaders`.

Reply:
Done in 6a8b10a. Resolving again. Thanks!